lib/spark.js

```diff
@@ -35,6 +39,94 @@ export default class Spark {
     return retrieval
   }
 
+  async executeRetrievalCheck (retrieval, stats) {
+    const provider = await this.queryTheIndex(retrieval, stats)
+    if (!provider) return
+    stats.protocol = provider.protocol
+    stats.providerAddress = provider.providerAddress
```

Comment:
why does `queryTheIndex` update some stats itself, but not these?

Reply:
I think it would be cleaner if we had

```js
const { provider, indexerResult } = await queryTheIndex(retrieval)
stats.indexerResult = indexerResult
```

This way the function has less side effects. It took me a bit to figure out which data was needed to be returned, and which modified

Reply:
Agreed, I was thinking about this too.